### ISSUE TITLE
Poetry shell - Problem when erasing accented letters

### ISSUE THREAD

**Issue by Biganon** [python-poetry/poetry#3327]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **Archlinux, 5.9.3-arch1-1**
- **Poetry 1.1.4**
- **termite v15**
- **zsh 5.8**
- No dependencies whatsoever

## Issue

Hi,

I create this very simple test script:

```python
print("Enter something: ", end="")
text = input()
print(text)
```

When I run it from my normal shell (zsh 5.8) I can type "ééé", and then erase those three characters, end of the story.

When I run it from inside poetry shell, I can type "ééé", and then I can erase all the way to the "g" at the end of something, including it. So I can "erode" the prompt, so it looks like "Enter somethin". If I now type something, with or without accents, it will be written right after "somethin", and it will be correctly registered and printed to me. So I can read "Enter somethinhello", hit Enter, and "hello" gets printed on the next line.

The number of prompt characters I can erase depends on the number of accented letters I've entered first. If I enter four of them, such as "éééé", I can erase one character further, until it reads "Enter somethi".

**Comment by sinoroc** [comment]
I can recreate the issue with:

* Ubuntu 18.04
* Gnome Terminal 3.28.2
* bash 4.4.20
* Poetry 1.1.4

**Comment by finswimmer** [comment]
I have no solution for you, but I'm fascinated by this problem :) Most probably it's in encoding problem. But why does this happen only in a shell created by poetry :thinking:

**Comment by sinoroc** [comment]
@finswimmer No clue either. Shells/terminals are quite tricky to tame. Did you manage to recreate the issue?

**Comment by finswimmer** [comment]
Yes, I'm able to reproduce it, if I start the script with `poetry run ...` or run `poetry shell` first. But I'm not able if I activate the venv with `source .venv/bin/activate`. So it has something to do how the shell is created.

**Comment by dimbleby** [comment]
this looks like a pexpect problem, maybe the same as https://github.com/pexpect/pexpect/issues/689

As there, you can recreate the issue inside the shell obtained through
```
python -c "import pexpect; c = pexpect.spawn('/bin/bash'); c.interact(); c.close()"
```

**Comment by dimbleby** [comment]
cf #4216
